### DISCUSSION TITLE
supabase/ssr: Offer other means of storing/sending session data other than cookies

### DISCUSSION THREAD

**Discussion by JorensM** [supabase/supabase#20567] [Feature Requests]
Hello, I'm writing an app with React Native and Expo, and after setting up `@supabase/ssr` I found out that React Native/Expo don't offer means of storing cookies. So I basically can't use `@supabase/ssr` on the client side because of this, because currently the package stores and sends session data using cookies and doesn't offer any other ways. It would be nice if there were some alternatives to storing/sending session data that would be compatible with React Native and Expo.

Thank you very much!

**Comment by charislam** [comment]
Hello @JorensM! I'm not super familiar with React Native, but if memory serves me, it's a client-side framework, right? Which means you wouldn't need to worry about SSR at all. You would instead need to hook up the basic Supabase Client from `@supabase/supabase-js` to use React Native's storage model instead. Does this guide work for you?

https://supabase.com/docs/guides/auth/quickstarts/react-native

**Reply by JorensM** [reply]
@charislam thanks for your response. The reason I need `supabase/ssr` is because I also have a backend and need to do some stuff there as well. So I need to be able to authenticate as the user on the backend.

**Reply by charislam** [reply]
Ahhh, I see. Makes sense.

**Comment by trevorpfiz** [comment]
@JorensM did you figure this out? I have been referencing https://github.com/supabase-community/create-t3-turbo, but it uses the `auth-helpers` and I was hoping to upgrade to `@supabase/ssr`.

**Reply by JorensM** [reply]
I stopped working on the project that needed this, but I think what you can do is create a custom client-to-server handler that securely sends the user's access token from the client to the backend and uses the token to do stuff on behalf of the user. I don't know how to make it secure though, you'll have to figure that out.
